### PR TITLE
cleaning up the clusterUpGKE

### PR DESCRIPTION
**Pull request by leiyiz**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
replace hard-coded options with options and env vars

```release-note
NONE
```

### CODE REVIEW

**Comment by leiyiz** [comment]
@saikat-royc

File: test/k8s-integration/cluster.go [hunk 1]
```diff
@@ -292,28 +292,42 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string, use
 			return fmt.Errorf("failed to create a new service: %v", err)
 		}
 
-		// TODO: change to read CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER env var
-		service.BasePath = "https://test-container.sandbox.googleapis.com/" // e.g test, staging env URL
+		service.BasePath = os.Getenv("CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER")
```
Comment (saikat-royc) [inline test/k8s-integration/cluster.go:295]: note: for prod, the env variable will be unset. All though we only have test jobs for test and staging env. If we want to run jobs for prod, this maynot work.

Comment (leiyiz) [reply test/k8s-integration/cluster.go:295]: oh I see... what's the endpoint for prod? I think a if statement here would be fine

Comment (saikat-royc) [reply test/k8s-integration/cluster.go:295]: it must be "container.sandbox.googleapis.com" i think. can you check in your setup?

**Comment by saikat-royc** [comment]
@leiyiz can you check the logic of generating skip string and handle node version skew (i.e when master and node pool are at different versions)? https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/main.go#L549

**Comment by saikat-royc** [comment]
overall changes look good. one nit: can you print the gcloud cluster list output right before waitForNodeDaemonset() so that we know exactly what cluster version was provisioned

File: test/k8s-integration/main.go [hunk 2]
```diff
@@ -424,27 +422,23 @@ func generateGCETestSkip(testParams *testParameters) string {
...
+		nodeVer = mustParseVersion(testParams.nodeVersion)
 	}
 
 	// Skip fsgroup change policy based on GKE cluster version.
-	if curVer.lessThan(mustParseVersion("1.20.0")) {
+	if curVer.lessThan(mustParseVersion("1.20.0")) || (nodeVer != nil && nodeVer.lessThan(mustParseVersion("1.20.0"))) {
 		skipString = skipString + "|fsgroupchangepolicy"
 	}
 
 	// Running snapshot tests on GKE needs k8s 1.19x storage e2e testsuite (until GKE supports v1 snapshot APIs).
```
Comment (saikat-royc) [inline test/k8s-integration/main.go:435]: Running snapshot tests on GKE clusters which does not support v1 snapshot CRDs needs 1.19x storage e2e testsuites...

File: test/k8s-integration/main.go [hunk 3]
```diff
@@ -424,27 +422,23 @@ func generateGCETestSkip(testParams *testParameters) string {
...
-	if curVer.lessThan(mustParseVersion("1.16.0")) {
-		skipString = skipString + "|allowExpansion"
+	var nodeVer *version
+	if testParams.nodeVersion != "" {
+		nodeVer = mustParseVersion(testParams.nodeVersion)
 	}
 
 	// Skip fsgroup change policy based on GKE cluster version.
-	if curVer.lessThan(mustParseVersion("1.20.0")) {
+	if curVer.lessThan(mustParseVersion("1.20.0")) || (nodeVer != nil && nodeVer.lessThan(mustParseVersion("1.20.0"))) {
```
Comment (saikat-royc) [inline test/k8s-integration/main.go:431]: we need a similar handling on 1.x.y-gke.z for useStagingDriver here?

**Pull request merged**